crates/noirc_driver/src/lib.rs

```diff
@@ -40,7 +41,7 @@ pub struct CompileOptions {
     pub deny_warnings: bool,
 
     /// Display output of `println` statements
-    #[arg(long)]
+    #[arg(long, default_value_if("prover_name", ArgPredicate::IsPresent, "true"))]
```

Comment:
Instead of changing the default here, I think we should likely remove it as an option entirely from `prove` and `execute`. Since it is meant to mirror `cargo test`'s show-output flag when running tests and cargo doesn't have it for other options either.